### DISCUSSION TITLE
Why Termux doesn't use Android system binaries from /bin?

### DISCUSSION THREAD

**Discussion by korpe1** [termux/termux-app#3803] [General]
AFAIK Android gives read access to /bin, etc, so why termux doesn't use them? For example /bin/ls seems well-working binary, but Termux uses /data/data/com.termux[...]/ls.

**Comment by agnostic-apollo** [comment]
The `/system/bin` core binaries are primarily provided by `toybox` and lot of them do not exist on older android versions or lack features or are severely broken. Termux core binaries are provided by `GNU` `coreutils` and they are the same for all android versions via termux packages repo, and have latest features and bugs get fixed with updates and are same as what linux distros like `debian` provide so that termux has a consistent environment with them.

https://android.googlesource.com/platform/system/core/+/master/shell_and_utilities/README.md

https://landley.net/toybox/

**Reply by korpe1** [reply]
Okay, anyway, can I just add Android system binaries to my $PATH?

**Reply by Biswa96** [reply]
adding /system/bin in $PATH may interfere with binaries in termux packages. You can run the program with full path. For example, `/system/bin/ls`

**Reply by korpe1** [reply]
> adding /system/bin in $PATH may interfere with binaries in termux packages. You can run the program with full path. For example, `/system/bin/ls`

What can it interfere?

**Reply by Biswa96** [reply]
agonstic-apollo mentioned that. for example, compare the help output of any programs - the /system/bin ones have less features and command line options. Also, dependencies may not have all functions exported - imagine running programs from LTS distributions in rolling release distribution.

**Reply by korpe1** [reply]
> agonstic-apollo mentioned that. for example, compare the help output of any programs - the /system/bin ones have less features and command line options. Also, dependencies may not have all functions exported - imagine running programs from LTS distributions in rolling release distribution.

Yeah, but e.g. `toybox ls` isn't dependency for any binary, right? I understand, if I run a script from Termux repos that depends spesific version of e.g. grep command it can break, but I think binaries of Termux should be just fine.

**Reply by Biswa96** [reply]
Yeah, the fact that _things break or not_ depends on the use cases.

**Reply by korpe1** [reply]
Yeah, but please give me an example of what can breaks if I add /system/bin to my $PATH? What are the use cases you refered to in your comment? I think executable binary programs doesn't depend on each other at all. I think it can break only shell scripts.

**Reply by korpe1** [reply]
ping @Biswa96
